openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -1217,3 +1217,24 @@ def component_version_asset(request, component_version_uuid, asset_path):
         content.read_file().chunks(),
         headers=redirect_response.headers,
     )
+
+
+@require_safe
+def component_draft_asset(request, usage_key_str, asset_path):
+    """
+    Serves the draft version of static assets associated with a Library Component.
+
+    See `component_version_asset` for more details
+    """
+    try:
+        key = UsageKeyV2.from_string(usage_key_str)
+        learning_package = authoring.get_learning_package_by_key(key.lib_key)
+        component = api.get_component_from_usage_key(key)
+        publishable_entity = authoring.get_publishable_entity_by_key(learning_package.id, component.key)
+        component_version_uuid = authoring.get_draft_version(publishable_entity.id).uuid
+    except InvalidKeyError as exc:
+        raise Http404() from exc
```

Comment:
Please make the invalid key a 400 (bad user request), and not a 404. Sometimes people just put in the wrong kind of key, or a malformed one because of some weird URL-escaping issue. We want to make sure they can distinguish between "it's not there" and "the key you sent us was malformed"